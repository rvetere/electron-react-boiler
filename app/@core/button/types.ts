import { SyntheticEvent } from 'react'

// TODO ...props sollte entfernt werden und die erlaubten Eigenschaften explizit hier aufgefÃ¼hrt werden
export interface IButtonProps {
  type?: string
  loading?: boolean
  primary?: boolean
  secondary?: boolean
  small?: boolean
  outline?: boolean
  big?: boolean
  inverted?: boolean
  block?: boolean
  mobileNoBlock?: boolean
  fullSize?: boolean
  disabled?: boolean
  title?: string
  tabIndex?: number
  onClick?: (event: SyntheticEvent) => void
  onMouseDown?: (event: SyntheticEvent) => void
  onMouseEnter?: () => void
  onMouseLeave?: () => void
}

export interface IButtonStylings {
  className: string
  otherProps: object
}
