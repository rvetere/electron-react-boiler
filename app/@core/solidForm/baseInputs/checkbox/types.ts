import { IValidationDisplayConfig, IValidationMessage } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { LabelPosition } from '@core/solidForm/types'
import react, { ReactNode } from 'react'

export interface ICheckboxProps extends react.InputHTMLAttributes<HTMLInputElement> {
  name: string
  inverted?: boolean
  labelPosition?: LabelPosition
  inputProps: IFormControllerInputProps
  value?: never
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
}

export interface ICheckboxState {
  isChecked?: boolean
  isValid: boolean
  isTouched: boolean
  validationMessages: IValidationMessage[]
}
