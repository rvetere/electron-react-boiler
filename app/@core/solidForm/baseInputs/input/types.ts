import { InputValue, ISelectOption } from '@core/solidForm'
import { IValidationDisplayConfig, IValidationMessage } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { IInputDefaultState } from '@core/solidForm/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import React, { ReactNode } from 'react'

export interface IInputTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  inverted?: boolean
  label?: ReactNode
  labelAppendix?: ReactNode
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
}

export type IInputTextState = IInputDefaultState

export interface IInputValidationPayload {
  value?: InputValue | ISelectOption[]
  validators?: ValidatorFn[]
  inputRef?: React.RefObject<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>
}

export interface IInputValidationResult {
  isValid: boolean
  messages: IValidationMessage[]
}
