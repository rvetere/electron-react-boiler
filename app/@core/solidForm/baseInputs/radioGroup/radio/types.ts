import { IRadioProps } from '@core/solidForm/baseInputs/radioGroup/types'
import { LabelPosition } from '@core/solidForm/types'
import react from 'react'

export interface IRadioOptionProps extends react.InputHTMLAttributes<HTMLInputElement> {
  value: string
  inverted?: boolean
  radioProps: IRadioProps
  labelPosition?: LabelPosition
}
