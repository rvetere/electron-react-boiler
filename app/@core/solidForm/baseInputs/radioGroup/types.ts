import { InputValue } from '@core/solidForm'
import { IValidationDisplayConfig } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { IInputDefaultState } from '@core/solidForm/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import react, { ReactNode } from 'react'

export interface IRadioGroupProps extends react.InputHTMLAttributes<HTMLInputElement> {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  initialValue?: string
  validationMessage?: string | null
  children: (_: IRadioProps) => ReactNode
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
  sideBySide?: boolean
}

export type IRadioGroupState = IInputDefaultState

export interface IRadioProps {
  updateRadioGroup: (value: string, initialValue?: string) => void
  initialValueRadioGroup: (value: string) => void
  name: string
  radioGroupValue?: InputValue
  sideBySide: boolean
  hasError: boolean
  describedById: string
}
