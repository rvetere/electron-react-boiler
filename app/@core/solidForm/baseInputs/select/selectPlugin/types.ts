import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import { ReactNode, SyntheticEvent } from 'react'
import { ISelectOption } from '../types'

export interface ISelectPluginProps {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  options?: ISelectOption[]
  value?: ISelectOption[] | string | number | undefined
  initialValue?: ISelectOption | ISelectOption[]
  onChange: (value: ISelectOption[]) => void
  onBlur: (event: SyntheticEvent) => void
  hasError?: boolean | null
  validationMessage?: string | null
  required?: boolean
  placeholder?: string
  describedById: string

  backspaceToRemoveMessage?: string
  multi?: boolean
  multiCheckbox?: boolean
  clearable?: boolean
  optionComponent?: ReactNode
  removeSelected?: boolean
  closeOnSelect?: boolean
  optionClassName?: string
  arrowRenderer?: ReactNode
  clearRenderer?: ReactNode
  noResultsText?: ReactNode
  loadingPlaceholder?: ReactNode
  // tslint:disable-next-line:no-any
  loadOptions?: (input?: string) => Promise<any>
}

export type IOptionsType<OptionType> = OptionType[]

export interface IGroupType<OptionType> {
  options: IOptionsType<OptionType>
  // tslint:disable-next-line:no-any
  [key: string]: any
}
