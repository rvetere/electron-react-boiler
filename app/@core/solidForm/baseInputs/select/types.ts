import { IValidationDisplayConfig, IValidationMessage } from '@core/solidForm/elements/infoSlot/validationDisplay/types'
import { IFormControllerInputProps } from '@core/solidForm/formController/types'
import { ValidatorFn } from '@core/solidForm/validators/types'
import react, { ReactNode } from 'react'
import { Omit } from 'utility-types'

type HTMLSelectPropsWithoutValue = Omit<react.InputHTMLAttributes<HTMLSelectElement>, 'value'>
type HTMLSelectPropsWithoutOnChageAndValue = Omit<HTMLSelectPropsWithoutValue, 'onChange'>

export interface ISelectOption {
  label: string | JSX.Element
  value: string | number
  disabled?: boolean
  template?: ReactNode
  clearableValue?: boolean
}
export interface ISelectProps extends HTMLSelectPropsWithoutOnChageAndValue {
  name: string
  inputProps: IFormControllerInputProps
  validators?: ValidatorFn[]
  inverted?: boolean
  label?: ReactNode
  labelAppendix?: ReactNode
  options?: ISelectOption[]
  value?: ISelectOption | ISelectOption[]
  multi?: boolean
  multiCheckbox?: boolean
  native?: boolean
  validationMessage?: string | null
  // tslint:disable-next-line:no-any
  loadOptions?: (input?: string) => Promise<any>
  onChange?: (value: ISelectOption | ISelectOption[]) => void
  hideValidation?: boolean
  infoSlot?: (validationConfig: IValidationDisplayConfig) => ReactNode
}

export interface ISelectState {
  value?: ISelectOption | ISelectOption[]
  isValid: boolean
  isTouched: boolean
  validationMessages: IValidationMessage[]
}

export interface ISelectValidationPayload {
  value: ISelectOption
  validators: ValidatorFn[]
  inputRef: React.RefObject<HTMLInputElement>
}
