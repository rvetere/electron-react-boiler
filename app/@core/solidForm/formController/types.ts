import { ReactNode } from 'react'
import { ISelectOption } from '../baseInputs/select/types'

export interface IFormControllerProps {
  className?: string
  hideWrappingFormTag?: boolean
  children: (actions: IFormControllerActions) => ReactNode
  // tslint:disable-next-line:no-any
  onSubmit?: (values: IFormValues, resetForm: FormResetFn) => Promise<any> | any
  onInvalid?: () => void
  onUpdate?: (values: IFormValues) => void
  onReset?: () => void
}

export type FieldValue = string[] | string | number | boolean | ISelectOption | ISelectOption[]

export interface IFormControllerState {
  fields: IFormControllerFields
  isLoading: boolean
}

export interface IFormValues {
  [name: string]: FieldValue
}

export interface IFormControllerFields {
  [name: string]: IFormControllerField
}

export type FormSubmitFn = () => void

export type FormResetFn = () => void

export interface IFormControllerActions {
  inputProps: IFormControllerInputProps
  submitForm: FormSubmitFn
  resetForm: FormResetFn
  formState: IFormControllerState
}

export interface IFormControllerInputProps {
  updateForm: (payload: IFormControllerPayload) => void
  registerInput: (payload: IFormControllerPayload) => void
  deRegisterInput: (payload: IFormControllerPayload) => void
}

export interface IFormControllerField {
  isValid: boolean
  value?: FieldValue
  reset: () => void
  touch: () => void
  focus: () => void
}

export interface IFormControllerPayload extends IFormControllerField {
  name: string
}

export type CurryUpdateState = ((field: IFormControllerState) => Pick<IFormControllerState, 'fields'>)
