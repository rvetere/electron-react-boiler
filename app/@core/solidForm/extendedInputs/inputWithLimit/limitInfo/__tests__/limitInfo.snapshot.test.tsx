/* eslint-env jest */
import { cleanup, render } from 'react-testing-library'
import { LimitInfo } from '..'
import { ILimitInfoProps } from '../types'

describe('LimitInfo', () => {
  afterEach(cleanup)
  const mockProps: ILimitInfoProps = {
    limit: 3,
    valueLength: 5
  }

  test('LimitInfo component should render correctly', () => {
    const { container } = render(<LimitInfo {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
