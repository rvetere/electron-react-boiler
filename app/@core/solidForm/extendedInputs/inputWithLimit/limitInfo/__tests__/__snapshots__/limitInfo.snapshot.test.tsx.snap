// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimitInfo LimitInfo component should render correctly 1`] = `
<div>
  <p
    class="limitInfo limitReached"
  >
    [0] / [1] characters
  </p>
</div>
`;
